tools/tpm2_tool.c

```diff
@@ -61,18 +62,101 @@ static ESYS_CONTEXT* ctx_init(TSS2_TCTI_CONTEXT *tcti_ctx) {
     return esys_ctx;
 }
 
+/*
+ * Build a list of the TPM2 tools linked into this executable
+ */
+static const tpm2_tool ** tools;
+static unsigned tool_count;
+
+void tpm2_tool_register(const tpm2_tool * tool)
+{
+    if (!tools)
+        tools = calloc(1024, sizeof(*tools));
+
+    //printf("%s: register %s\n", __func__, tool->name);
+    if (tools)
+        tools[tool_count++] = tool;
```

Comment:
This could overflow if someone added a bazillion tools and didn't update this, make sure bounds checking is done and it complains loudly.

For extra points we could calculate the tools size in the Makefile and pass this in so the list is just the right size and we don't have waste.